### ISSUE TITLE
How to load a new image to scale for this year's challenge?

### ISSUE THREAD

**Issue by cghercoias** [Andrew-Schwartz/Pathing#1]
Hi Andrew - Mr. G here.
How do you create a new image to scale for this year's challenge?
There are three challenges in autonomous mode (pictures attached).
![Barrel_Racing_path](https://user-images.githubusercontent.com/8409018/107860789-656ae080-6e0f-11eb-8378-487569e41fe1.png)
![bounce_path](https://user-images.githubusercontent.com/8409018/107860790-66037700-6e0f-11eb-927f-62bbc27e2b74.png)
![slalompath](https://user-images.githubusercontent.com/8409018/107860791-669c0d80-6e0f-11eb-826a-6b7d5a01fffa.png)

Where do we load them? Do they go in \src\main\resources\images ?

I did generate a csv (see attached).

[Barrel_Racing_save.zip](https://github.com/Andrew-Schwartz/Pathing/files/5976540/Barrel_Racing_save.zip)

Then what do we do with the .csv files, where do they get loaded in the main robot code?

Many thanks,
Mr. G.

**Comment by Annie-Schwartz** [comment]
Hello Mr. Ghercoias!
The way I intended for a new image to be set was through a picker in the program's options: go to `Files -> Set Image`, and find the new background image in the normal file explorer. This works while the program is open, however, I apparently never implemented saving which image to use between uses of the program. The "easiest" fix would just be to include the the field image in the `src/main/resources/images/` folder, and update the path to the file in line 90 of `UIController.java` (the path should still start with `images/`). The "better" fix, if someone on the team wants to write a bit of new code for this, would be to make the `backgroundImage` field (same java file & line) not hardcoded but instead call `Config.getStringProperty` with some key like `"image_path"`, and then call `Config.setProperty` with the new image path in the method in `UIController.java` that handles picking images (should be similar code to what the `.

As for the csv files, (assuming that the wpilib project structure is the same as it was two years ago), those should go in the `src/main/deploy/` directory in the code that is deployed to the robot. I remember the method I used for saving paths was a bit strange: the way it works was the control points needed to calculate a path can be saved separately from the csv that contains the motor displacements/velocities that robot would follow. This was necessary because generating the path on the fly could take time away from the start of auto, but only saving the motor's instructions didn't leave any way for the path to be reconstructed by anything other than the robot actually running. To load the csv files when the robot code is ran, they should be in the `/home/lvuser/deploy/` directory of the RoboRio (files in the `src/main/deploy/` directory get copied there when code is uploaded to the robot). I had a repo that has some preliminary code to follow paths, I think it's private right now but I could make it public if you'd like. 

Hope that helped,
Andrew :)
